src/user-guide/writing-workflows/scheduling.rst

```diff
@@ -2015,22 +2015,25 @@ tools to limit workflow activity.
 Runahead Limiting
 ^^^^^^^^^^^^^^^^^
 
-The scheduler runahead limit determines how many consecutive cycle points can
-be active at once. The base point of the runahead calculation is the current
-lowest-valued point with :term:`active` or :term:`incomplete` tasks present.
+The runahead limit prevents a workflow from getting too far ahead of the oldest
+active cycle point.
 
-The runahead limit can be set with :cylc:conf:`[scheduling]runahead limit`,
-to an integer interval (for datetime or integer cycling workflows) or to
-a datetime interval (datetime cycling only).
+A cycle is considered to be active if it contains any :term:`active` or
+:term:`incomplete` tasks.
```

Comment:
We need to get our story straight on exactly what "active cycle points" (aka the "active window" of the workflow) contain(s):
- literally active tasks (submitted, running)
- incomplete tasks (finished without completing required outputs)
- waiting tasks that are ready to run as far as the task graph is concerned, but are still waiting on
  - clock triggers
  - xtriggers
  - queue release
  - or hold

This isn't the right place to say all of that, but we should put it somewhere and link all references to "active cycle points" to it.

Reply:
What about tasks with partially satisfied dependencies?
I assume an active cycle is one with a task in the n=0 window?

Reply:
Uh, yes! Especially after the "Cylc set" PR (which ditches the hidden pool)

Reply:
Ok, it's sounding like we need a new glossary entry to describe the meaning of "active cycle".

The definition should include:
* Active tasks.
* Incomplete tasks.
* Waiting tasks that have one or more of their dependencies satisfied.
* Waiting parentless tasks.

Good?

Reply:
I think we should define an active cycle as one which contains a task in the active ("n=0") window.
This term is already defined in the glossary but is wrong - I think we should update it to include:
* Active tasks
* Incomplete tasks
* Active-waiting tasks (we might need to tweak the existing definition in the glossary for these)
* Waiting tasks with some, but not all, of their pre-requisites met

> Waiting parentless tasks

I don't understand the relevance of parentless here.

Reply:
Plus (as I said above):
- waiting tasks that are ready to run as far as the task graph is concerned, but are still waiting on: queues, hold, clock and xtriggers

> >    Waiting parentless tasks
> 
> I don't understand the relevance of parentless here.

Parentless tasks get auto-spawned to the runahead limit (although, after @dwsutherland 's extant PR clock-triggered ones won't). That is technically covered by "ready to run as far at the task graph is concerned" but we should document it anyway - might not be obvious to users.